foundationdb/src/tuple.rs

```diff
@@ -8,20 +8,81 @@
 
 //! Tuple Key type like that of other FoundationDB libraries
 
-use byteorder::{self, ByteOrder};
 use std;
 use std::io::Write;
 
-#[derive(Debug, Clone, Copy)]
+use byteorder::{self, ByteOrder};
+
+#[derive(Debug, Fail)]
 pub enum TupleError {
-    //
+    #[fail(display = "Unexpected end of file")]
     EOF,
-    InvalidType,
+    #[fail(display = "Invalid type")]
+    InvalidType(u8),
+    #[fail(display = "Invalid data")]
     InvalidData,
 }
 
 type Result<T> = std::result::Result<T, TupleError>;
 
+/// Various tuple types
+#[derive(Clone, Copy, Eq, PartialEq)]
+enum SingleType {
+    Empty = 0x00,
+    Bytes = 0x01,
+    String = 0x02,
+    Nested = 0x05,
+    IntZero = 0x14,
+    PosIntEnd = 0x1d,
+    NegIntStart = 0x0b,
+    Float = 0x20,
+    Double = 0x21,
+    False = 0x26,
+    True = 0x27,
+    Uuid = 0x30,
+    VersionStamp = 0x33,
+}
+
+impl SingleType {
+    /// Tries to convert a u8 into a SingleType
+    fn try_from(value: u8) -> Result<Self> {
```

Comment:
How about implementing `std::convert::TryFrom` trait?
https://doc.rust-lang.org/std/convert/trait.TryFrom.html

Reply:
TryFrom is nightly only right now, is it not?

Reply:
I thought it landed on stable, but not yet as yoy told. Sorry for confusion.